docs/guides/db-connection-pooling.md

```diff
@@ -0,0 +1,151 @@
+---
+id: db-connection-pooling
+title: Database Connection Pooling
+sidebar_label: Database Connection Pooling
+---
+
+Opening a database connection is an expensive process, and that's why we use **[Connection pool](https://en.wikipedia.org/wiki/Connection_pool)** to reduce the cost. And it also allows us to have a good transaction management in a single session that uses different providers.
+
+The example uses PostgreSQL and **[node-postgres](https://node-postgres.com/features/transactions)** below;
```

Comment:
Mention here that this examples uses postgres, but you can do it with any modern database (for example, with mongo: https://www.compose.com/articles/connection-pooling-with-mongodb/)